### ISSUE TITLE
Removing finalizers from mgmt cluster breaks the UI

### ISSUE THREAD

**Issue by a-blender** [rancher/dashboard#9393]
**Setup**
- Rancher version: 2.7-head
- Rancher UI Extensions:
- Browser type & version: Mac Safari

**Describe the bug**

If finalizers on the management cluster are removed, deleting the management cluster while the provisioning cluster object still exists the dashboard hangs. Trying to access the `Cluster Management` tab gives me a `null undefined id` error or reloads to another page location.

**To Reproduce**

* Provision a single node RKE2 node driver cluster via the rancher UI
* Delete the mgmt cluster (or remove finalizers)
* Navigate to Cluster Management / around the UI

**Result**

**Expected Result**

**Screenshots**

**Additional context**

**Comment by gaktive** [comment]
@Andy-Blendermann thanks for finding this. Is this a blocker for Q3 or can this wait for Q4?

**Comment by richard-cox** [comment]
I'm not sure deleting the mgmt cluster whilst the provisioning cluster exists is a supported case. I'd guess it should be the other way around? We would need someone on the backend to confirm

**Comment by a-blender** [comment]
@gaktive Nope, not a blocker! Just something that took down our dev environment UI when I force removed a cluster. @jakefhyde is removing the prov cluster first, then mgmt the expected process if someone has to, for some reason, force remove a stuck cluster?

**Comment by gaktive** [comment]
@Andy-Blendermann cool, thanks for clarifying! We'll target Q4 at the earliest.

**Comment by jakefhyde** [comment]
It's not expected at all from a backend perspective, but we should handle it gracefully if possible, as currently it becomes much harder to navigate to a specific cluster to fix the issue. It's definitely something a user may run into when trying to clean up a cluster manually (which I believe is how @a-blender encountered the issue initially).

**Comment by eva-vashkevich** [comment]
We should check if it still happens after the changes done to Cluster list in 2.15
